lib/rpc/src/transport/smoltcp.rs

```diff
@@ -19,7 +20,7 @@ const RX_BUF_LEN: usize = 8192;
 const TX_BUF_LEN: usize = 8192;
 
 pub struct TCPTransport<'a> {
-    iface: RefCell<Interface<'a, DevQueuePhy>>,
+    iface: Arc<RefCell<Interface<'a, DevQueuePhy>>>,
```

Comment:
This isn't for this PR but longer-term: we should think about what to do with the device Interface since RefCell is only for a single core.
cc @prestonsn this might be a good place to start looking into the NR+device interface model without having to implement an NVMe device first but use a NIC device instead